src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -280,7 +305,7 @@ public bool TryPersist(OptionKey optionKey, object value)
                 }
             }
 
-            _settingManager.SetValueAsync(storageKey, value, isMachineLocal: false);
+            _brokeredSettingsManager.SetValueAsync(storageKey, value, CancellationToken.None);
```

Comment:
so... if this throws and nobody cares (note the lack of an await or any other follow-up), why call the method at all?

Reply:
What error handling do you propose?

Reply:
See my [write-up on how to handle such a situation](https://github.com/Microsoft/vs-threading/blob/master/doc/cookbook_vs.md#how-to-write-a-fire-and-forget-method-responsibly).

Reply:
There's a deeper question too perhaps for @JoshuaBStevens: so let's say that call _fails_, say because the network dropped out. What should we do? Do we need to retry? Will the local service handle any retries for a network drop for us?

Reply:
When Live Share is involved, network drops are handled in the lower transport layer: Live Share itself keeps track of which packets get ACK'd and if a connection drops, it will try to reestablish and retransmit the packets that didn't get ACK'd so that everyone at a higher level have no idea there was a disruption. If the connection can't be recovered, well, there's nothing you can do but throw.
But if the RPC service itself simply crashes or otherwise hangs up on you, you get an exception at the client right away.
But since you're talking to an always-local service, which in turn then talks to a remote service, the fact that you addressed the question to @JoshuaBStevens is apt. I just wanted to throw in some surrounding concerns so you know when you're the one talking to the remote service.

Reply:
We don't do anything with exceptions originating from remote calls; we just let them flow out.  @AArnott, would you recommend handling such exceptions by assuming that the connection to the server has ended?  I.e. proceed as we would in the "not connected" scenario?

Reply:
Given you've documented that the service is always local, yes gracefully handling the remoting exceptions and switching to local mode makes sense IMO so your callers don't have to worry so much about it. That said, some of your callers will be out of proc, so they may have *some* connection issues of their own to deal with. Or when you move out of proc yourself.

Reply:
I've opened a bug (1042829) to track handling this kind of exception internally as you suggest.